### ISSUE TITLE
Scrollbars and Safearea on iOS

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#25802]
Looking at an iOS native Scrollview (e.g. Safari or Mailbox in Mail) in landscape on an iPhone Xs you will see the following behavior:

* When the notch is on the left, the scrollbar is drawn all the way at the right edge of the screen (the content of the scrollview does not go all the way to the right to avoid potential obtrusion by the round corners)
* When the notch is on the right, the scrollbar is not drawn all the way to the right and instead slightly indented (just like the content) to avoid the notch

I believe I cannot easily replicate this behavior with the `Safearea` and `Scrollbar` widget in Flutter. Either the scrollbar is always (regardless of notch orientation) drawn with a padding from the edge of the screen (and would avoid the notch if the phone is orientated to have the notch on the right) or the scrollbar is always drawn all the way at the screen edge (and thus obstructed by the notch if it happened to be on the right side).

**Comment by matthew-carroll** [comment]
@goderbauer would you mind posting comparable screenshots showing what we do vs what the standard platform behavior looks like?

**Comment by tomgilder** [comment]
Here's the standard platform behaviour - will show what Flutter does later :-)

Note that this isn't just a landscape thing.

![iOS Scrollbar behavior](https://user-images.githubusercontent.com/756862/55933064-ad79b480-5c34-11e9-985b-dbe46c44e68d.gif)

**Comment by tomgilder** [comment]
Note that in the iOS example above, I didn't have to do anything - UIKit provides the behaviour for free by adjusting the [scrollIndicatorInsets](https://developer.apple.com/documentation/uikit/uiscrollview/1619427-scrollindicatorinsets) automatically.

This example wraps the content in a SafeArea:
![Flutter with SafeArea](https://user-images.githubusercontent.com/756862/55934830-94740200-5c3a-11e9-9bf9-7df468b64669.gif)

This one doesn't use a SafeArea, but has padding on the content:
![Flutter without SafeArea](https://user-images.githubusercontent.com/756862/55934829-93db6b80-5c3a-11e9-9c9b-39721c502d30.gif)

**Comment by matthew-carroll** [comment]
@tomgilder thanks for posting the examples. 

CC @xster @LongCatIsLooong

**Comment by xster** [comment]
Leaving this bug open but I think it's mostly a dupe of #13253 where accounting for the MediaQueryData.padding in the thumb paint calculation will solve.

**Comment by goderbauer** [comment]
I think the issue I had was slightly different. Look at the location of the scrollbar in safari: If there is no notch, the scrollbar is as far to the edge of the screen as possible. If there is a notch, the scrollbar is placed with a padding to avoid the notch:

![native](https://user-images.githubusercontent.com/1227763/56056059-fe121000-5d0f-11e9-9a20-d86bdf0c0187.gif)

As far as I can tell, there is no way to get that behavior easily in flutter. You either have a padding wether or not there is a notch, or there is never padding and the scrollbar goes through the notch. Example from Flutter gallery:

![flutter](https://user-images.githubusercontent.com/1227763/56055982-c73bfa00-5d0f-11e9-9b28-7be3a300ee7c.gif)

**Comment by xster** [comment]
You're right, I think there is an additional problem indeed. Looking at iOS, it seems like on opposite side of the notch in landscape mode actually has 2 paddings. The nav bar menus, list cell items, tab bar etc all have one padding and the scrollbar has another. 

<img width="1184" alt="Screen Shot 2019-04-15 at 9 17 35 PM" src="https://user-images.githubusercontent.com/156888/56181795-f7abbe80-5fc3-11e9-99c2-9658ddb815e5.png">

This could be tricky to handle since we already defined the `padding` and `viewInset` metrics and the framework doesn't have any additional information from the engine to split 2 different types of `padding` by reading https://developer.apple.com/documentation/uikit/uiview/2891103-safeareainsets.

**Comment by TahaTesser** [comment]
Hi @goderbauer 
Is this issue still valid?
Thank you

**Comment by tomgilder** [comment]
@TahaTesser Yes. There are definitely still issues on iPhones with notches on landscape mode, including Flutter drawing the safe area for the notch on the wrong side.

I think the issues with portrait are all fixed.

**Comment by TrGiLong** [comment]
Hi, @goderbauer 

This problem is also applied to horizontal scroll in portrait mode. You can see more detail in my closed PR #79314. The `Scrollbar` always calculates the indicator position with the value from all sides in `MediaQueryData.padding`. Correct me if I'm wrong because the PR was a long time ago.

If your scroll area in portrait mode is not full (set the height to 300 and centralize it), the scroll indicator position will also be wrong. The area of the scroll indicator is smaller since flutter adds screen padding.

I think we should know when we apply the padding. Is it possible in flutter to apply padding in depends on the position of the widget?

Otherwise, I set padding in `MediaQuery` to zero to avoid extra padding for scrollbars, which are inside of other widgets

**Comment by SwissCheese5** [comment]
This issue still happens in 2022. In a horizontal iPhone 13, the SafeArea thinks the notch is on the right side when it is on the left side and vice-versa.

**Comment by Shahmirzali-Huseynov** [comment]
any progress ?
